src/object/obj_verify.c

```diff
@@ -632,22 +632,173 @@ dc_obj_verify_cmp(struct dc_obj_verify_args *dova_a,
 	return 0;
 }
 
-int
-dc_obj_verify_rdg(struct dc_object *obj, struct dc_obj_verify_args *dova,
-		  uint32_t rdg_idx, uint32_t reps, daos_epoch_t epoch)
+static int
+dc_obj_verify_ec_cb(struct dss_enum_unpack_io *io, void *arg)
+{
+	struct dc_obj_verify_args	*dova = arg;
+	struct dc_object		*obj = obj_hdl2ptr(dova->oh);
+	d_sg_list_t			sgls[DSS_ENUM_UNPACK_MAX_IODS];
+	d_iov_t			 	iovs[DSS_ENUM_UNPACK_MAX_IODS] = { 0 };
+	d_sg_list_t			sgls_verify[DSS_ENUM_UNPACK_MAX_IODS];
+	d_iov_t			 	iovs_verify[DSS_ENUM_UNPACK_MAX_IODS] = { 0 };
+	tse_task_t			*task;
+	tse_task_t			*verify_task;
+	uint64_t			shard = dova->current_shard;
+	int				nr = io->ui_iods_top + 1;
+	int				i;
+	int				rc;
+
+	D_DEBUG(DB_TRACE, "compare "DF_KEY" nr %d shard "DF_U64"\n", DP_KEY(&io->ui_dkey),
+		nr, shard);
+	if (nr == 0)
+		return 0;
+
+	for (i = 0; i < nr; i++) {
+		daos_size_t	size;
+		char		*data;
+		char		*data_verify;
+		daos_iod_t	*iod = &io->ui_iods[i];
+
+		size = daos_iods_len(&io->ui_iods[i], 1);
+		D_ASSERT(size != -1);
+		D_ALLOC(data, size);
+		if (data == NULL)
+			D_GOTO(out, rc =-DER_NOMEM);
```

Comment:
(style)  spaces required around that '=' (ctx:WxO)
(style)  space required before that '-' (ctx:OxV)